code/go/0chain.net/validatorcore/storage/handler.go

```diff
@@ -2,23 +2,95 @@ package storage
 
 import (
 	"context"
+	"fmt"
 	"net/http"
 	"os"
 	"runtime/pprof"
+	"strings"
+	"time"
 
 	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"github.com/didip/tollbooth/v6"
+	"github.com/didip/tollbooth/v6/limiter"
+	"github.com/spf13/viper"
 
 	"github.com/gorilla/mux"
 )
 
+const (
+	RequestPerSecond      = 5
+	DefualtExpirationTime = time.Minute * 5
+)
+
+var lmt *limiter.Limiter
+
+func init() {
+	defaultIPLookups := []string{"RemoteAddr", "X-Forwarded-For", "X-Real-IP"}
+	lmt = tollbooth.NewLimiter(RequestPerSecond, &limiter.ExpirableOptions{
+		DefaultExpirationTTL: DefualtExpirationTime,
+	})
+
+	lmt.SetIgnoreURL(true)
+	lmt.SetIPLookups(defaultIPLookups)
+}
```

Comment:
The rate limit code in this file seems duplicate the work that we've done in common package. I can see why this is needed, but I would suggest to define functions in common to new single rate limit with configs passed in as param instead of reading from viper directly. In that way, we can reuse the rate limit code in common package.

Maybe we should also avoid global `limits` variables defined in common package, those variables should be created by the users with `NewRateLimit()` function and stored in their top level main packages and to be reused.